pkg/destroy/gcp/address.go

```diff
@@ -115,23 +80,56 @@ func (o *ClusterUninstaller) deleteAddress(ctx context.Context, item cloudResour
 // destroyAddresses removes all address resources that have a name prefixed
 // with the cluster's infra ID.
 func (o *ClusterUninstaller) destroyAddresses(ctx context.Context) error {
-	for _, scope := range []resourceScope{gcpGlobalResource, gcpRegionalResource} {
-		found, err := o.listAddresses(ctx, scope)
+	for _, ad := range []addressDestroyer{
+		{
+			itemTypeName: "address",
+			destroyFunc:  o.addressDelete,
+			listFunc:     o.addressList,
+		},
+		{
+			itemTypeName: "regionaddress",
+			destroyFunc:  o.regionAddressDelete,
+			listFunc:     o.regionAddressList,
+		},
```

Comment:
I think all this parameterizing is hurting readability if you end up changing almost everything for only 2 resource types. Would it be too much duplication to do instead:
```golang
func (o *ClusterUninstaller) destroyAddresses(ctx context.Context) error {
   items, err := o.findAndDeleteGlobalAddress(ctx)
   if err != nil {   
       errs = append(errs, err)
   }
   o.insertPendingItems(items)
   ...
   items, err := o.findAndDeleteRegionAddress(ctx)
   ...
}

func (o *ClusterUninstaller) findAndDeleteGlobalAddress(ctx context.Context) ([]cloudResource, error) {
    req := o.computeSvc.GlobalAddresses.List(o.ProjectID).Fields(googleapi.Field("items(name),nextPageToken")).Filter(o.clusterIDFilter())
	err := req.Pages(ctx, func(list *compute.AddressList) error {
		result = append(result, createAddressCloudResources(o.addressList, list)...)
		return nil
	})
	if err != nil {
		return nil, fmt.Errorf("failed to list global addresses: %w", err)
	}
	return result, nil
}
```

Reply:
Just example, not actually working code ^